pkg/verification/verify_request.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verification
+
+import (
+	"crypto"
+
+	"github.com/digitorus/timestamp"
+	"github.com/pkg/errors"
+)
+
+func VerifyRequest(ts *timestamp.Request) error {
+	// only SHA-1, SHA-256, SHA-384, and SHA-512 are supported by the underlying library
```

Comment:
```suggestion
	// only SHA-256, SHA-384, and SHA-512 are supported by the underlying library
```

Reply:
What I meant here was that we only need to handle SHA-1, since only these 4 algorithms are supported. Otherwise, we'd also need to check MD4/MD5, etc.

Reply:
Ah gotcha, that makes sense.